File: 期末考2.0/Q1 Student Inheritance/Date.cpp
#include"Date.h"

Date::Date(){}
Date::Date(int newYear, int newMonth, int newDay){
    year = newYear;
    month = newMonth;
    day = newDay;
}
void Date::setYear(int newYear){
    year = newYear;
}
void Date::setMonth(int newMonth){
    month = newMonth;
}
void Date::setDay(int newDay){
    day = newDay;
}
int Date::getYear(){
    return year;
}
int Date::getMonth(){
    return month;
}
int Date::getDay(){
    return day;
}

File: 期末考2.0/Q5 Python Student Inheritance/Date.py
class Date():
    def __init__(self, newYear, newMonth, newDay):
        self.__year = newYear
        self.__month = newMonth
        self.__day = newDay
    def setYear(self, newYear):
        self.__year = newYear
    def setMonth(self, newMonth):
        self.__month = newMonth
    def setDay(self, newDay):
        self.__day = newDay
    def getYear(self):
        return self.__year
    def getMonth(self):
        return self.__month
    def getDay(self):
        return self.__day
    def toString(self):
        print(str(self.getYear()) + "-" + str(self.getMonth()) + "-" + str(self.getDay()), end=' ')